permiso/src/main/java/com/greysonparrelli/permiso/PermisoDialogFragment.java

```diff
@@ -48,16 +55,102 @@ public static PermisoDialogFragment newInstance(@Nullable String title, @NonNull
         return dialogFragment;
     }
 
+    public static class Builder {
+        private int titleId = 0;
+        private String title = null;
+        private int buttonTextId = 0;
+        private String message = null;
+        private int messageId = 0;
+        // by default, interpret the dialog msg body as string text
+        private boolean interpretHtml = false;
+        private String buttonText = null;
+
+
+        private Builder() {}
+
+        public Builder(int titleId, int msgBody, int buttonTextId) {
+            this();
+            this.titleId = titleId;
+            this.messageId = msgBody;
+            this.buttonTextId = buttonTextId;
+        }
+
+
+        public int getTitleId() { return titleId; }
+        public String getTitle() { return title; }
```

Comment:
The original motivation for me submitting this feature request was because I wanted create a concrete implementation of Permiso.IOnPermissionResult (in a separate class) that does not require me to pass in an activity.   My use case for Permiso.IOnPermissionResult is one of data-only.  I would like to create an enum of dialog data and pass just that in.  Requiring getTitleId(Context) would bring back the dependency on the activity.
What do you think?